Panic when registering a route with an empty path

Hi,

Since 1.3.1 I can't register a route with an empty path anymore.

This used to work:

```
	adminGroup := r.Mount("/admin")
	adminGroup.HandleFunc("GET ", home)
```

But now it panics:

```
panic: parsing "/adminGET ": at offset 0: invalid method "/adminGET"

goroutine 1 [running]:
net/http.(*ServeMux).register(...)
        /usr/local/go/src/net/http/server.go:2797
net/http.(*ServeMux).HandleFunc(0xc0000ebd78?, {0xc000015330?, 0xa203e8?}, 0x9dfe56?)
        /usr/local/go/src/net/http/server.go:2771 +0x65
github.com/go-pkgz/routegroup.(*Bundle).register(0xc0000ebd78, {0x9dfe56, 0x4}, 0xa203e8)
        /home/hli/go/pkg/mod/github.com/go-pkgz/routegroup@v1.3.1/group.go:162 +0x1fd
github.com/go-pkgz/routegroup.(*Bundle).HandleFunc(...)
        /home/hli/go/pkg/mod/github.com/go-pkgz/routegroup@v1.3.1/group.go:119
git.epic.net/epic/toolbox/statuspage/pkg/handlers/admin.Mount(0xc0000f0b90)
        /home/hli/epic/toolbox/statuspage/pkg/handlers/admin/mux.go:13 +0x245
main.main()
        /home/hli/epic/toolbox/statuspage/main.go:49 +0x612
```

Reverting to 1.3.0 fixes it.

ServeMux's documentation indicates that this should be valid:

`[METHOD ][HOST]/[PATH]`

> All three parts are optional; "/" is a valid pattern. If METHOD is present, it must be followed by at least one space or tab.

If time permits I'll try and take a shot at fixing.

--- a/README.md
+++ b/README.md
@@ -99,6 +99,36 @@ You can also apply middleware to specific routes inside the group without modify
 apiGroup.With(corsMiddleware, apiMiddleware).Handle("GET /hello", helloHandler)
 ```
 
+**Handling Root Paths Without Trailing Slashes**
+
+When working with mounted groups, you often need to handle requests to the group's root path without a trailing slash. For this purpose, `routegroup` provides the `HandleRoot` method:
+
+```go
+// Create mounted groups
+apiGroup := router.Mount("/api")
+v1Group := apiGroup.Mount("/v1")
+usersGroup := v1Group.Mount("/users")
+
+// Handle the root paths (no trailing slashes)
+apiGroup.HandleRoot("GET", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+    // This handles requests to "/api" (without trailing slash)
+    w.Write([]byte("API Documentation"))
+}))
+
+usersGroup.HandleRoot("GET", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+    // This handles requests to "/api/v1/users" (without trailing slash)
+    w.Write([]byte("List users"))
+}))
+
+// Different HTTP methods can be handled separately
+usersGroup.HandleRoot("POST", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+    // This handles POST requests to "/api/v1/users"
+    w.Write([]byte("Create user"))
+}))
+```
+
+While it's also possible to handle such paths using a trailing slash pattern (`"/"`) with the regular `Handle` or `HandleFunc` methods, that approach results in a redirect from non-trailing slash URLs (e.g., `/api`) to the trailing slash version (e.g., `/api/`). The `HandleRoot` method avoids this redirect, providing a more direct response and avoiding an extra round-trip, which is especially important for non-GET requests or when clients don't automatically follow redirects.
+
 **Alternative Usage with `Route`**
 
 You can also use the `Route` method to add routes and middleware in a single function call:

--- a/group.go
+++ b/group.go
@@ -165,6 +165,27 @@ func (b *Bundle) register(pattern string, handler http.HandlerFunc) {
 // Route allows for configuring the Group inside the configureFn function.
 func (b *Bundle) Route(configureFn func(*Bundle)) { configureFn(b) }
 
+// HandleRoot adds a handler for the group's root path (no trailing slash).
+// This method is needed to handle requests to a group's root path without causing redirects that would occur with trailing slash patterns.
+// Unlike registering a pattern with "/", which causes the standard ServeMux to redirect non-trailing slash requests to the trailing slash
+// version (301 Moved Permanently), HandleRoot registers the exact path without any special treatment,
+// avoiding the redirect behavior while still applying middleware.
+// Method parameter can be empty; in this case, the handler will be registered for all methods.
+func (b *Bundle) HandleRoot(method string, handler http.Handler) {
+	// for empty base path, use "/" to match the root
+	pattern := b.basePath
+	if pattern == "" {
+		pattern = "/"
+	}
+
+	// add method if specified
+	if method != "" {
+		pattern = method + " " + pattern
+	}
+
+	b.mux.Handle(pattern, b.wrapMiddleware(handler))
+}
+
 // wrapMiddleware applies the registered middlewares to a handler.
 func (b *Bundle) wrapMiddleware(handler http.Handler) http.Handler {
 	for i := len(b.middlewares) - 1; i >= 0; i-- {

--- a/group_test.go
+++ b/group_test.go
@@ -2627,6 +2627,47 @@ func TestHandleRoot(t *testing.T) {
 		}
 	})
 
+	t.Run("HandleRoot with empty base path", func(t *testing.T) {
+		// create a group with empty base path
+		group := routegroup.New(http.NewServeMux())
+		group.Use(func(next http.Handler) http.Handler {
+			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("X-Middleware", "applied")
+				next.ServeHTTP(w, r)
+			})
+		})
+
+		// handle the root path (empty base path)
+		group.HandleRoot("GET", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if _, err := w.Write([]byte("root")); err != nil {
+				t.Fatalf("failed to write response: %v", err)
+			}
+		}))
+
+		ts := httptest.NewServer(group)
+		defer ts.Close()
+
+		// test GET request to root
+		resp, err := client.Get(ts.URL + "/")
+		if err != nil {
+			t.Fatalf("failed to make request: %v", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("expected status 200, got %d", resp.StatusCode)
+		}
+		if resp.Header.Get("X-Middleware") != "applied" {
+			t.Errorf("middleware not applied")
+		}
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("failed to read response body: %v", err)
+		}
+		if string(body) != "root" {
+			t.Errorf("expected 'root', got '%s'", body)
+		}
+	})
+
 	t.Run("handle with trailing slash", func(t *testing.T) {
 		group := routegroup.New(http.NewServeMux())
 		apiGroup := group.Mount("/api")

--- a/group_test.go
+++ b/group_test.go
@@ -2478,6 +2478,186 @@ func TestSubgroupRootPathMatching(t *testing.T) {
 	})
 }
 
+func TestHandleRoot(t *testing.T) {
+	// create client that doesn't follow redirects
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse // don't follow redirects
+		},
+	}
+
+	t.Run("HandleRoot with middleware", func(t *testing.T) {
+		group := routegroup.New(http.NewServeMux())
+		group.Mount("/api").Route(func(apiGroup *routegroup.Bundle) {
+			apiGroup.Use(func(next http.Handler) http.Handler {
+				return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("X-Middleware", "applied")
+					next.ServeHTTP(w, r)
+				})
+			})
+			apiGroup.HandleRoot("GET", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if _, err := w.Write([]byte("api root")); err != nil {
+					t.Fatalf("failed to write response: %v", err)
+				}
+			}))
+			apiGroup.HandleFunc("GET /test", func(w http.ResponseWriter, r *http.Request) {
+				if _, err := w.Write([]byte("test")); err != nil {
+					t.Fatalf("failed to write response: %v", err)
+				}
+			})
+		})
+
+		ts := httptest.NewServer(group)
+		defer ts.Close()
+
+		// test direct access to registered root /api - should NOT redirect and middleware should be applied
+		resp, err := client.Get(ts.URL + "/api")
+		if err != nil {
+			t.Fatalf("failed to make request: %v", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("expected status 200, got %d", resp.StatusCode)
+		}
+		if resp.Header.Get("X-Middleware") != "applied" {
+			t.Errorf("middleware not applied")
+		}
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("failed to read response body: %v", err)
+		}
+		if string(body) != "api root" {
+			t.Errorf("expected 'api root', got '%s'", body)
+		}
+
+		// test access to /api/test
+		resp, err = client.Get(ts.URL + "/api/test")
+		if err != nil {
+			t.Fatalf("failed to make request: %v", err)
+		}
+		defer resp.Body.Close()
+		body, err = io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("failed to read response body: %v", err)
+		}
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("expected status 200, got %d", resp.StatusCode)
+		}
+		if string(body) != "test" {
+			t.Errorf("expected 'test', got '%s'", body)
+		}
+
+		// test POST request to /api
+		req, err := http.NewRequest(http.MethodPost, ts.URL+"/api", http.NoBody)
+		if err != nil {
+			t.Fatalf("failed to create request: %v", err)
+		}
+		resp, err = client.Do(req)
+		if err != nil {
+			t.Fatalf("failed to make request: %v", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusNotFound {
+			t.Errorf("expected status 404, got %d", resp.StatusCode)
+		}
+	})
+
+	t.Run("HandleRoot without method", func(t *testing.T) {
+		group := routegroup.New(http.NewServeMux())
+		group.Mount("/data").Route(func(dataGroup *routegroup.Bundle) {
+			dataGroup.Use(func(next http.Handler) http.Handler {
+				return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("X-Middleware", "applied")
+					next.ServeHTTP(w, r)
+				})
+			})
+			// register without specifying a method (empty string)
+			dataGroup.HandleRoot("", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if _, err := w.Write([]byte("data root")); err != nil {
+					t.Fatalf("failed to write response: %v", err)
+				}
+			}))
+		})
+
+		ts := httptest.NewServer(group)
+		defer ts.Close()
+
+		// test GET request
+		resp, err := client.Get(ts.URL + "/data")
+		if err != nil {
+			t.Fatalf("failed to make request: %v", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("expected status 200, got %d", resp.StatusCode)
+		}
+		if resp.Header.Get("X-Middleware") != "applied" {
+			t.Errorf("middleware not applied")
+		}
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("failed to read response body: %v", err)
+		}
+		if string(body) != "data root" {
+			t.Errorf("expected 'data root', got '%s'", body)
+		}
+
+		// test POST request - should also work since no method was specified
+		req, err := http.NewRequest(http.MethodPost, ts.URL+"/data", http.NoBody)
+		if err != nil {
+			t.Fatalf("failed to create request: %v", err)
+		}
+		resp, err = client.Do(req)
+		if err != nil {
+			t.Fatalf("failed to make request: %v", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("expected status 200, got %d", resp.StatusCode)
+		}
+		if resp.Header.Get("X-Middleware") != "applied" {
+			t.Errorf("middleware not applied")
+		}
+		body, err = io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("failed to read response body: %v", err)
+		}
+		if string(body) != "data root" {
+			t.Errorf("expected 'data root', got '%s'", body)
+		}
+	})
+
+	t.Run("handle with trailing slash", func(t *testing.T) {
+		group := routegroup.New(http.NewServeMux())
+		apiGroup := group.Mount("/api")
+		apiGroup.HandleFunc("GET /", func(w http.ResponseWriter, r *http.Request) {
+			if _, err := w.Write([]byte("api root")); err != nil {
+				t.Fatalf("failed to write response: %v", err)
+			}
+		})
+
+		ts := httptest.NewServer(group)
+		defer ts.Close()
+
+		resp, err := client.Get(ts.URL + "/api")
+		if err != nil {
+			t.Fatalf("failed to make request: %v", err)
+		}
+		defer resp.Body.Close()
+
+		// verify trailing slash approach causes redirect
+		if resp.StatusCode != http.StatusMovedPermanently {
+			t.Errorf("expected redirect status 301, got %d", resp.StatusCode)
+		}
+
+		location := resp.Header.Get("Location")
+		if location != "/api/" {
+			t.Errorf("expected redirect to '/api/', got '%s'", location)
+		}
+	})
+
+}
+
 func ExampleNew() {
 	group := routegroup.New(http.NewServeMux())
 
@@ -2553,4 +2733,30 @@ func ExampleBundle_Route() {
 	if err := http.ListenAndServe(":8080", group); err != nil {
 		panic(err)
 	}
+}
+
+// This example shows how to use HandleRoot to handle the root path of a mounted group without trailing slash
+func ExampleBundle_HandleRoot() {
+	group := routegroup.New(http.NewServeMux())
+
+	// create API group
+	apiGroup := group.Mount("/api")
+
+	// apply middleware
+	apiGroup.Use(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("X-API", "true")
+			next.ServeHTTP(w, r)
+		})
+	})
+
+	// handle root path (responds to /api without redirect)
+	apiGroup.HandleRoot("GET", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "API root")
+	}))
+
+	// regular routes
+	apiGroup.HandleFunc("GET /users", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "List of users")
+	})
 }

